assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentResultArchiveUploader.kt

```diff
@@ -208,13 +197,31 @@ abstract class AssessmentResultArchiveUploader(
     }
 }
 
-fun ArchiveFile.fileInfoMap() : Map<String, JsonElement> {
-    val map = mutableMapOf(
-        "filename" to JsonPrimitive(filename),
-        "timestamp" to JsonPrimitive(endDate.toString())
-    )
+fun ArchiveFile.fileInfo(): ArchiveFileInfo {
     if (this is JsonArchiveFile) {
-        map.put("contentType", JsonPrimitive("application/json"))
+        return ArchiveFileInfo(filename, endDate.toString(), "application/json")
+    } else {
+        return ArchiveFileInfo(filename, endDate.toString())
     }
-    return map
-}
\ No newline at end of file
+}
+
+@Serializable
+data class MetadataMap(
+    val taskIdentifier: String,
+    val deviceInfo: String,
+    val appName: String,
+    val appVersion: String,
+    val deviceTypeIdentifier: String,
+    val taskRunUUID: String,
+    val dataGroups: String,
+    val startDate: kotlinx.datetime.Instant,
+    val endDate: kotlinx.datetime.Instant?,
+    val files: List<ArchiveFileInfo>
+)
+
+@Serializable
+data class ArchiveFileInfo(
+    val filename: String,
+    val timestamp: String,
+    val contentType: String? = null
```

Comment:
```suggestion
    val contentType: String? = null
    val identifier: String? = null
    val stepPath: String? = null
    val jsonSchema: String? = null
```

This would match the Swift model:
```swift
public struct FileInfo : Codable, Hashable, Equatable {

    /// The filename of the archive object. This should be unique within the manifest. It may include
    /// a relative path that points to a subdirectory.
    public let filename: String
    
    /// The file creation date.
    public let timestamp: Date
    
    /// The content type of the file.
    public let contentType: String?
    
    /// The identifier for the result. This value may *not* be unique if a step is run more than once
    /// during a task at different stages.
    public let identifier: String?
    
    /// The full path to the result if it is within the step history.
    public let stepPath: String?
    
    /// The uri for the json schema if the content type is "application/json".
    public let jsonSchema: URL?
}
```